src/gui/editors/SongEditor.cpp

```diff
@@ -108,8 +193,10 @@ SongEditor::SongEditor( Song * song ) :
 			this, SLOT( selectRegionFromPixels( int, int ) ) );
 	connect( m_timeLine, SIGNAL( selectionFinished() ),
 			 this, SLOT( stopRubberBand() ) );
+	connect( m_song, SIGNAL( playbackStateChanged() ),
+			 this, SLOT( playbackStateChanged() ) );
 
-	m_positionLine = new positionLine( this );
+	m_positionLine = new positionLine( this, m_zoomingModel, m_song );
```

Comment:
Instead of passing `m_zoomingModel` to the position line, consider listening for its `dataChanged` signal and then updating the position line's width.

Reply:
I'm already listening for the signal, I use the attribute as a way to get the zooming value.

Reply:
You could pass the zooming value as a parameter to `positionLine::zoomUpdate`. In fact, this would allow you to pass `m_zoomLevels[m_zoomingModel->value()]`, which means you don't need to duplicate `m_zoomLevels` in `positionLine`.

Reply:
Well, I've removed `zoomUpdate()` because it was redundant (I connected the signal to `update()`). I'll rewrite it though.

Reply:
Is it acceptable to keep it this way or do I need a rewrite?